### DISCUSSION TITLE
Model lighting in AR mode

### DISCUSSION THREAD

**Discussion by ttvettuparambil** [google/model-viewer#3525] [General]
Is there any method of controlling the lighting of the model when the 3D model is in AR mode. Suppose part of the model is in sunlight, can that part have a lighter shade than the rest of the model?

**Comment by elalish** [comment]
Use https://modelviewer.dev/docs/index.html#entrydocs-lightingandenv-attributes-environmentImage to change the lighting for both 3D and WebXR AR. If you want AR mode to use dynamic lighting estimation from your camera, use https://modelviewer.dev/docs/index.html#entrydocs-augmentedreality-attributes-xrEnvironment, but it'll only work in WebXR mode and it doesn't always look good. I believe Scene Viewer and Quick Look do this always.

**Reply by ttvettuparambil** [reply]
Thank you for the quick response.

My query is if we place the 3D AR model near a light source, based on the natural lighting of the environment can parts of the model striking the light have lighter color and other parts have darker shade...currently the lighting changes based on the distance between the camera and the 3D AR model.

Tried the xr-Envionment example link today...placed the astronaut model in such a way that half of the model is in sunlight and the other half is in shadow...did not seem to work for me.
https://modelviewer.dev/examples/augmentedreality/#ar

I found a documentation in official AR core website
https://developers.google.com/ar/develop/lighting-estimation/#hdr_cubemap
This is the requirement that I am looking to fulfil using model viewer.

**Reply by elalish** [reply]
Please try each different AR mode and record screen captures so we can see the differences. I think it may be working, but just not very well. Lighting estimation, especially on Android, still has a ways to go in quality.

**Comment by ttvettuparambil** [comment]
Thank you for your quick response.

I recently installed an AR plugin in WordPress that uses model-viewer to render AR models. When the model is shown in AR mode, it seems that the lighting is affected by the distance of the camera from the model instead of natural lighting. I have recorded a video regarding this which is present in the below link.

https://drive.google.com/file/d/1EpXBZxYkbXcdflq-Gmxu_anVlE4E_HYH/view?usp=sharing

Also I tried the dynamic lighting estimation examples from the link that you sent me and none of them seem to be working. I used the canoe 3d model instead of the astronaut model  because the model seemed to have better lighting as well as reflection so thought that it would work better.

I would like to know if dynamic lighting estimation works as what I have recorded. I am new to 3D and AR that's why I have so many doubts.

Many thanks,
Thomas

**Reply by elalish** [reply]
Thanks for the video! I don't think the issue is how close the camera is to the model, but what the lighting estimator understands of the scene based on your camera's motion. What you're seeing is the imperfections of AR lighting estimation algorithms.
